layers/Ptensors0.hpp

```diff
@@ -330,8 +330,9 @@ namespace ptens{
     template<typename OUTPUT>
     void add_gather_back(const OUTPUT& x, const LayerMap& map){
       auto plan=GatherPlanFactory::gather_map0(map,x.atoms,atoms,x.getk(),0);
-      if constexpr(std::is_same<OUTPUT,Ptensors0<TYPE> >::value)
+      if constexpr(std::is_same<OUTPUT,Ptensors0<TYPE> >::value){
```

Comment:
This is where the work actually happens on the C++ side.
I am not quite sure the gather plan works, that is what confuses me.

Here is what I found out so far.

When we enter this member function, the object is numerically but of the right dimension for the output.
And x is as expected the `g_view` value that we put in.

The in plan.out() is: `[ 0 1 1 1 0 ]` and the plan.in() is `[ 0 1 0 1 0 ]`.

I checked the value of the result of `x.reduce0(plan.out()`. and is the same numerical 0 as before `[ 1.70958e-43 1.96462e-42 1.13505e-43 0 ]`.

And after the broadcast0 call the this object is still numerically zero.